src/src/pages/capabilities/jsonmetadata/index.js

```diff
@@ -0,0 +1,203 @@
+      setCurrentMetadataString(prettyMetadata);
+      setJsonString(prettyMetadata);
+    }
+  }, [metadata]);
+
+  useEffect(() => {
+    if (schemaString === "") {
+      selfServiceApiClient.getCapabilityJsonMetadataSchema().then((schema) => {
+        setSchemaString(prettifyJsonString(schema));
+      });
+    }
+  }, []);
+
+  const checkIfJsonIsParsable = (json) => {
+    try {
+      JSON.parse(json);
+      return true;
+    } catch (exception) {
+      setValidationError("Invalid JSON: " + exception.message);
+      return false;
+    }
+  };
+
+  const checkIfFollowsSchema = (json) => {
+    try {
+      const ajv = new Ajv();
+      const validate = ajv.compile(JSON.parse(schemaString));
+      const valid = validate(JSON.parse(json));
+      if (!valid) {
+        setValidationError("Schema Error: " + validate.errors[0].message);
+      }
+      return valid;
+    } catch (exception) {
+      setValidationError("Schema Error: " + exception.message);
+      return false;
+    }
+  };
+
+  const isJsonValid = (json) => {
+    return validationError === "";
+  };
+
+  const prettifyJsonString = (json) => {
+    return JSON.stringify(JSON.parse(json), null, 2);
+  };
+
+  const submitJsonMetadata = async () => {
+    const prettyJsonString = prettifyJsonString(jsonString);
+    if (isDirty) {
+      setCapabilityJsonMetadata(prettyJsonString);
+      setCurrentMetadataString(prettyJsonString);
+      setJsonString(prettyJsonString);
+    }
+    setIsEditing(false);
+  };
+
+  const startEditing = () => {
+    setIsEditing(true);
+  };
+  const cancelEditing = () => {
+    setIsEditing(false);
+    setJsonString(currentMetadataString);
+  };
+
+  return (
+    <>
+      <PageSection headline="Metadata">
+        {!isEditing && (
+          <div>
+            <div className={styles.metadataView}>
+              <SyntaxHighlighter
+                language="json"
+                style={syntaxStyle}
+                wrapLongLines={false}
+                customStyle={{
+                  margin: "0",
+                }}
+              >
+                {currentMetadataString}
+              </SyntaxHighlighter>
+            </div>
+            <ButtonStack>
+              <Button size="small" variation="primary" onClick={startEditing}>
+                Edit
+              </Button>
+            </ButtonStack>
+          </div>
+        )}
+        {isEditing && (
+          <div>
+            <div className={styles.jsonParent}>
+              <div className={styles.jsonInputParent}>
+                <MonacoEditor
+                  name={"jsonmetadata"}
+                  language="json"
+                  value={jsonString}
+                  errorMessage={validationError}
+                  onChange={(e) => {
+                    setIsDirty(true);
+                    setJsonString(e);
+                    if (!checkIfJsonIsParsable(e)) return;
+                    if (!checkIfFollowsSchema(e)) return;
```

Comment:
Could we not use the `isJsonValid` function for clarity?

Reply:
these functions also set the error message so not really...I am open for suggestions how to rewrite it.